src/app/unlocked-stage/unlocked-stage.component.ts

```diff
@@ -1,156 +1,331 @@
-import { Component, OnInit, AfterViewChecked, Renderer } from '@angular/core';
+import { Component, HostListener, OnInit, Renderer2 } from '@angular/core';
```

Comment:
I think it will be better to define constants for the hardcoded values wherever possible in this file.

Reply:
Okay, done that.